cli/sawtooth_cli/block.py

```diff
@@ -94,50 +91,52 @@ def print_yaml(data):
         keys = ('num', 'block_id', 'batches', 'txns', 'signer')
         headers = (k.upper() if k != 'batches' else 'BATS' for k in keys)
 
-        def get_block_data(block):
+        def get_data(block):
             batches = block.get('batches', [])
-            txn_count = reduce(
-                lambda t, b: t + len(b.get('transactions', [])),
-                batches,
-                0)
-
+            txns = [t for b in batches for t in b.get('transactions', [])]
             return (
                 block['header'].get('block_num', 0),
                 block['header_signature'],
                 len(batches),
-                txn_count,
-                block['header']['signer_pubkey']
-            )
+                len(txns),
+                block['header']['signer_pubkey'])
 
         if args.format == 'default':
-            # Fit within 150 chars, without truncating block id
-            print('{:<3}  {:128.128}  {:<4}  {:<4}  {:11.11}'.format(*headers))
+            # Set column widths based on window and data size
+            _, window_width = os.popen('stty size', 'r').read().split()
```

Comment:
We should retain Windows support for the CLI.

We should also avoid using the tty size when the output is being redirected to a file "sawtooth block list > foo", as well as running without a tty such as with ssh: "ssh localhost sawtooth block list", in which case stty gives different error output based on platform.  sys.stdout.isatty() might be useful for this purpose.